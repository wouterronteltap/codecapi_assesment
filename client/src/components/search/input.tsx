import React, { useState, useEffect } from 'react';
import { TextField } from '@material-ui/core';
import { History } from 'history';

import { useDebounce } from "../../hooks/useDebounce";
import { CONSTANTS } from "../../utils/contants";

interface IProps {
    search: string,
    history: History,
    urlquery: string
}

const SearchInput = React.memo((props:IProps) => {
    const { history, search, urlquery } = props;
    const [query, setQuery] = useState('');
    const debounced = useDebounce(query, 300);
    useEffect(() => {
        if(debounced){
            history.push(CONSTANTS.QUERY_TEMPLATE(debounced));
        }
    }, [debounced]);
    useEffect(() => {
        if(search){
            history.push(search);
        }
    }, []);
    return(
        <TextField
            type={'search'}
            label={'Search user'}
            variant={'outlined'}
            value={urlquery}
            onChange={e => setQuery(e.target.value)}
            placeholder={'Search GitHub user'}
            />
    )
});

export { SearchInput }