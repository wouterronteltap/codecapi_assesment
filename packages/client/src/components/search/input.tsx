import React, { useState, useEffect } from "react";
import { TextField } from "@material-ui/core";

import { useAppDispatch } from "../../context";
import { useDebounce } from "../../hooks/useDebounce";

const SearchInput = React.memo(({ lastquery = "" }: { lastquery: string }) => {
  const { dispatch } = useAppDispatch();
  const [query, setQuery] = useState(lastquery);
  const debounced = useDebounce(query, 300);
  useEffect(() => {
    if (debounced) {
      dispatch({
        type: "SET_DEBOUNCED_QUERY",
        payload: {
          query: debounced
        }
      });
    }
  }, [debounced]);
  return (
    <TextField
      type={"search"}
      label={"Search user"}
      variant={"outlined"}
      value={query}
      onChange={e => setQuery(e.target.value)}
      placeholder={"Search GitHub user"}
    />
  );
});

export { SearchInput };
