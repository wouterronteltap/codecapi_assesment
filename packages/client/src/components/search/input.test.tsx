import React from "react";
import { render, cleanup } from "@testing-library/react";
import { SearchInput } from "./input";

describe("The SearchInput Component", () => {
  afterEach(cleanup);
  test("renders correctly", () => {
    const { asFragment } = render(<SearchInput lastquery={""} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("should be initialized with custom lastquery", () => {
    const query = "wouter";
    const component = render(<SearchInput lastquery={query} />);
    const input: HTMLInputElement = component.getByPlaceholderText(
      "Search GitHub user"
    ) as HTMLInputElement;
    expect(input.value).toBe(query);
  });
});
