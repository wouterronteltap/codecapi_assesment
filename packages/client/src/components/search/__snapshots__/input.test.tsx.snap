// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SearchInput Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink="false"
    >
      Search user
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputTypeSearch"
        placeholder="Search GitHub user"
        type="search"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-61 MuiOutlinedInput-notchedOutline"
        style="padding-left: 8px;"
      >
        <legend
          class="PrivateNotchedOutline-legend-62"
          style="width: 0.01px;"
        >
          <span>
            â€‹
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
